bulk.go

```diff
@@ -0,0 +1,117 @@
+package opensearchtools
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/opensearch-project/opensearch-go/v2"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchapi"
+)
+
+// BulkRequest wraps the functionality of [opensearchapi.Bulk] by supporting request body creation.
+// An empty BulkRequest will fail to execute. At least one Action is required to be added.
+//
+// For more details see https://opensearch.org/docs/latest/api-reference/document-apis/bulk/
+type BulkRequest struct {
+	// Actions lists the actions to be performed in the BulkRequest
+	Actions []BulkAction
+
+	// Refresh determines if the request should wait for a refresh or not
+	Refresh Refresh
+
+	// Index determines the entire index for the request
+	Index string
+}
+
+// NewBulkRequest instantiates an empty BulkRequest
+func NewBulkRequest() *BulkRequest {
+	return &BulkRequest{}
+}
+
+// Add an action to the BulkRequest.
+func (r *BulkRequest) Add(actions ...BulkAction) *BulkRequest {
+	r.Actions = append(r.Actions, actions...)
+	return r
+}
+
+// SetIndex on the request
+func (r *BulkRequest) SetIndex(index string) *BulkRequest {
```

Comment:
I think `WithIndex` would make more sense here since it's sort of a builder style where you're returning the receiver.